src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/FromFhirToEdifactMapper.java

```diff
@@ -8,5 +8,20 @@
         return parameters -> segment;
     }
 
+    static FromFhirToEdifactMapper<?> optional(OptionalFromFhirToEdifactMapper<?> mapper, Parameters parameters) {
+        if (mapper.canMap(parameters)) {
+            return mapper;
+        }
+        return new FromFhirToEdifactMapper.EmptyMapper();
+    }
+
     T map(Parameters parameters);
+
+    class EmptyMapper implements FromFhirToEdifactMapper<Segment> {
+
+        @Override
+        public Segment map(Parameters parameters) {
+            return null;
```

Comment:
why empty mapper in this case returns null?

Reply:
Nulls are filtered in translators